pyfarm/agent/http/api/assign.py

```diff
@@ -81,6 +81,16 @@ def post(self, **kwargs):
             request.finish()
             return NOT_DONE_YET
 
+        if self.agent.reannounce_lock.locked:
+            logger.warning("Temporarily rejecting assignment because we "
+                           "are in the middle of a reannounce.")
+            request.setResponseCode(BAD_REQUEST)
+            request.write(
+                dumps({"error": "Agent cannot accept assignments because of a "
+                                "reannounce in progress. Try again shortly."}))
+            request.finish()
+            return NOT_DONE_YET
```

Comment:
It would be cleaner and more efficient to just return the body and response code:

```python
return dumps({"error": "Agent cannot accept assignments because of a "
                       "reannounce in progress. Try again shortly."}), BAD_REQUEST
```

However, there's a bug in the base resource that would essentially prevent `BAD_REQUEST` from being set.  #282 should fix this bug though.  I don't think it's worth holding back this change till that fixed is merge though given what this PR is fixing.